tests/todo.spec.js

```diff
@@ -0,0 +1,139 @@
+    await expect(page).toHaveTitle(/Todo/);
+  })
+  
+  test('check if footer buttons redirect to correct page and have correct text', async({page}) => {
+    let testpage = new playwrightPage(page);
+  
+    await expect(testpage.createdByLink).toHaveText('Remo H. Jansen');
+    await testpage.createdByLink.click();
+    await expect(page).toHaveURL('https://github.com/remojansen/');
+  
+    await testpage.accessPage();
+  
+    await expect(testpage.partOfLink).toHaveText('TodoMVC');
+    await testpage.partOfLink.click();
+    await expect(page).toHaveURL('https://todomvc.com/');
+  })
+  
+  test('check if header button redirects to correct page and have correct text', async({page}) => {
+    let testpage = new playwrightPage(page);
+  
+    await expect(testpage.headerLink).toHaveText('real TodoMVC app.');
+    await testpage.headerLink.click();
+    await expect(page).toHaveURL('https://todomvc.com/');
+  })
+})
+
+test.describe('add and remove tasks', () => {
+  test('check if tasks can be added in todo list', async ({page}) => {
+    let testpage = new playwrightPage(page); 
+    await testpage.insertItemsInTodoList();
+
+    const todoItems = await testpage.getTodoItems();
+    for(let item of todoItems)
+      await expect(item).toBeVisible();
+  }); 
+  
+  test('check if tasks can be deleted', async ({page}) => {
+    let testpage = new playwrightPage(page);
+  
+    await testpage.insertItemsInTodoList();
+    const itemCountBeforeDelete = await testpage.todoItemList.count();
+  
+    await testpage.clickFirstDeleteButton();
+  
+    const itemCountAfterDelete = await testpage.todoItemList.count();
+    expect(itemCountAfterDelete).toBeLessThan(itemCountBeforeDelete);
+  });
+})
+
+test.describe('check footer of todo modal', () => {
+  test('check nr of items added in todoList', async({page}) => {
+    let testpage = new playwrightPage(page);
+    await testpage.insertItemsInTodoList();
+    
+    const taskCounter = await testpage.todoItemList.count();
+    const itemsleft = parseInt((await testpage.nrOfItemsInTodoList.textContent()),10);
+    
+    expect(itemsleft).toBe(taskCounter);
+  });
+  
+  test('check if active section shows correct tasks', async ({page}) => {
+    let testpage = new playwrightPage(page);
+  
+    await testpage.insertItemsInTodoList();
+    const allTasksCounter = await testpage.todoItemList.count();
+    await testpage.checkBoxItem.check();
+    await testpage.activeButton.click();
+    const activeTaskCounter = await testpage.todoItemList.count();
+    expect(activeTaskCounter).toBeLessThan(allTasksCounter);
+  })
+  
+  test('check if completed section shows correct tasks', async ({page}) => {
+    let testpage = new playwrightPage(page);
+  
+    await testpage.insertItemsInTodoList();
+    await testpage.checkBoxItem.check();
+    await testpage.completedButton.click();
+    const completedTaskCounter = await testpage.todoItemList.count();
+    expect(completedTaskCounter).toBe(1);
+  })
```

Comment:
test title and test assertion is not matching. what exactly you want to test ?

Reply:
at the footer of todo modal we have three buttons "All" "Active" and "Completed" so I want to check if marking one task as done, it is moved to "Completed" section